test/files/mock-rhsm-rest

```diff
@@ -0,0 +1,62 @@
+#! /usr/bin/python3
+
+from http.server import HTTPServer, BaseHTTPRequestHandler
+import re
+import ssl
+import os
+import shutil
+import sys
+import time
+
+
+class handler(BaseHTTPRequestHandler):
+    def match(self, p):
+        return re.fullmatch(p, self.path)
+
+    def do_POST(self):
+        data = self.rfile.read(int(self.headers['Content-Length']))
+        m = self.match("/auth/realms/redhat-external/protocol/openid-connect/token")
+        if m and data == b'grant_type=refresh_token&client_id=rhsm-api&refresh_token=my_offline_token':
+            self.send_response(200)
+            self.end_headers()
+            self.wfile.write(b'{ "access_token": "my_access_token" }\n')
+            return
+
+        self.send_response(404)
+        self.end_headers()
+
+    def do_GET(self):
+        authorization = self.headers['Authorization']
+        m = self.match("/management/v1/images/rhel/.*/.*")
+
+        if m and authorization == 'Bearer my_access_token':
+            self.send_response(200)
+            self.end_headers()
+            self.wfile.write(b'{ "body": [\
+                {"imageName": "RHEL 5.0 Boot ISO", "filename": "rhel-5-0-boot.iso",\
+                "downloadHref": "https://api.access.redhat.com/management/v1/images/my_image_checksum/download"}\
+            ] }\n')
+            return
+
+        m = self.match("/management/v1/images/my_image_checksum/download")
+        if m and authorization == 'Bearer my_access_token':
+            filepath = "/var/lib/libvirt/images/example.img"
+            with open(filepath, 'rb') as f:
+                self.send_response(200)
+                self.send_header("Content-Type", 'application/octet-stream')
+                self.send_header("Content-Disposition", f"attachment; filename='{os.path.basename(filepath)}'")
+                fs = os.fstat(f.fileno())
+                self.send_header("Content-Length", str(fs.st_size))
+                self.end_headers()
+                time.sleep(3)  # Simulate network delay
+                shutil.copyfileobj(f, self.wfile, 1024)  # Specify 1KiB buffer size, otherwise the download might be too fast on some platforms
```

Comment:
This now might be too slow -- as long as this takes at least 2 seconds, it's fine for the tests. But as long as it's < 10s, it's fine (i.e. we shouldn't make our tests unnecessarily slow)

Reply:
Lowered, now it's around 2-3 seconds